### PR TITLE
chore: initial setup

### PR DESCRIPTION
**Pull request by Hitesh-Kandpal**
**Description:**

Initial setup for the project, added linting and workflow as mentioned in the coding standard

**Related changes:**

N/A

**Screenshots:**

Screenshot 2024-04-30 at 12 14 36 PM

**Pending actions:**

N/A

**Additional notes:**

N/A

## Summary by CodeRabbit

- **New Features**
  - Introduced GitHub Actions workflows for automated linting and building of both frontend and backend components.
  - Added editor configurations, ESLint, and Prettier settings to ensure consistent code formatting and quality.
  - Configured Visual Studio Code settings and tasks for enhanced development experience.
  - Added new Angular components, modules, and routing configurations to streamline app development.
  - Provided comprehensive project documentation including setup, building, and testing instructions.

- **Documentation**
  - Updated README files with detailed project setup and command instructions.

- **Chores**
  - Updated `.gitignore` files across multiple directories to exclude unnecessary files from version control.

### CODE REVIEW

File: portal/.prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (xixas) [inline portal/.prettierrc.json:1]: Remove the cache files. Add it git ignore.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Standard Angular App Flow
```
Comment (xixas) [inline .github/workflows/main.yml:1]: Try to follow the folder structure as used for osmox(we have apps folder).

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18'  # Adjust this to the Node.js version your project needs
```
[the remark this answers is not in the document]
Comment (xixas) [reply .github/workflows/main.yml:18]: Try to understand the structure for mono repo like osmox and create files like that.

File: portal/.vscode/extensions.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (xixas) [inline portal/.vscode/extensions.json:1]: Are all these  .vscode files recommended as per the angular standards documentation?

File: portal/src/app/app.routes.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import { Routes } from '@angular/router';
```
Comment (xixas) [inline portal/src/app/app.routes.ts:1]: I am not sure about this usecase of angular 17 as I have not yet tried. Assuming this is correct way for now.

File: .github/workflows/main.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Hitesh-Kandpal) [reply .github/workflows/main.yml]: update it to 20, and created files like osmox

File: portal/.prettierrc.json (hunk 1 above)
Comment (Hitesh-Kandpal) [reply portal/.prettierrc.json]: there is already a gitignore file in portal like osmox, added a gitignore file in root directory just like the one in osmox

File: .github/workflows/main.yml (hunk 2 above)
Comment (Hitesh-Kandpal) [reply .github/workflows/main.yml]: fixed it, foloowed osmox structure

File: portal/.vscode/extensions.json (hunk 4 above)
Comment (Hitesh-Kandpal) [reply portal/.vscode/extensions.json]: in angular standard, nothing like this was mentioned, there the document only said that change/modify setting.json files, and these files in vscode was autogenerated and they are also in osmox project so I kept it

File: portal/src/app/app.routes.ts (hunk 5 above)
Comment (Hitesh-Kandpal) [reply portal/src/app/app.routes.ts]: this was beacause of standalone, this is now fixed, we are using routing module and app module just like osmox

**Pull request merged**